app/controllers/ArrivalMessagesController.scala

```diff
@@ -39,28 +39,33 @@ class ArrivalMessagesController @Inject()(cc: ControllerComponents,
                                    messageConnector: MessageConnector,
                                    xmlValidationService: XmlValidationService)(implicit ec: ExecutionContext) extends BackendController(cc) with HttpErrorFunctions {
 
-  def createUnloadingPermission(arrivalId: String): Action[NodeSeq] = authAction.async(parse.xml) {
+  def sendMessageDownstream(arrivalId: String): Action[NodeSeq] = authAction.async(parse.xml) {
     implicit request =>
-      xmlValidationService.validate(request.body.toString, UnloadingRemarksXSD) match {
-        case Right(_) =>
-          messageConnector.post(request.body.toString, arrivalId).map { response =>
-            response.status match {
-              case s if is2xx(s) =>
-                response.header(LOCATION) match {
-                  case Some(locationValue) => Utils.arrivalId(locationValue, fragmentIndex = -2) match {
-                    case Success(id) =>
-                      Accepted.withHeaders(LOCATION -> s"/customs/transits/movements/arrivals/${Utils.urlEncode(id)}/messages")
-                    case Failure(_) =>
-                      InternalServerError
-                  }
-                  case _ =>
-                    InternalServerError
+      request.body.head.label match {
+        case "CC044A" =>
```

Comment:
I'm really not a fan of this, partially for being a magic string (it'd be nice to have it form part of the XSD type as a field) but it'd be nicer to be able to just match against the parse result like you showed me `XSDSuccessful(type) match` or something like that.